ui/src/org/theGraphApi.ts

```diff
@@ -61,6 +63,74 @@ export interface Org {
   projectCount?: number;
 }
 
+export interface Safe {
+  id: string;
+  owners: string[];
+  threshold: number;
+}
+
+export async function getSafesByOwner(owners: string[]): Promise<Safe[]> {
+  const safesResponse = await orgsSubgraphClient().query<{
+    safes: Array<{
+      // Safe address.
+      id: string;
+      // Safe owner addresses.
+      owners: string[];
+      // Number of signatures required for quorum.
+      threshold: string;
+    }>;
+  }>({
+    query: apolloCore.gql`
+      query GetSafesByOwners($owners: [String!]!) {
+        safes(where: { owners_contains: $owners }) {
+          id
+          owners
+          threshold
+        }
+      }
+      `,
+    variables: { owners: owners.map(owner => owner.toLowerCase()) },
+  });
+
+  return safesResponse.data.safes.map(safe => ({
+    ...safe,
+    threshold: Number(safe.threshold),
+  }));
+}
+
+export const getSafeMetadata = memoizeLru(
+  async (address: string): Promise<Safe> => {
+    const safesResponse = await orgsSubgraphClient().query<{
+      safe: {
+        // Safe address.
+        id: string;
+        // Safe owner addresses.
+        owners: string[];
+        // Number of signatures required for quorum.
+        threshold: string;
+      };
+    }>({
+      query: apolloCore.gql`
+      query GetSafe($address: ID!) {
+        safe(id: $address) {
+          id
+          owners
+          threshold
+        }
+      }
+      `,
+      variables: { address: address.toLowerCase() },
+    });
+
+    return {
+      ...safesResponse.data.safe,
+      threshold: Number(safesResponse.data.safe.threshold),
+    };
+  },
+  safeAddress => safeAddress,
+  { max: 1000, maxAge: 15 * 60 * 1000 } // TTL 15 minutes
```

Comment:
Is it reasonable to memoize this? What are the downsides of having a much shorter ttl, say 20 seconds?

Reply:
I think it still makes sense to memoize it because we're getting a list of all the orgs and the associated metadata on the org explore screen. Maybe we could lower it to something like 5min, so that the org explore page doesn't take forever to load?

Reply:
Also the Graph queries cost us in production.

Reply:
Ok. Let’s keep it like this then.